core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultMemoryHealthChecker.java

```diff
@@ -0,0 +1,44 @@
+package com.linecorp.armeria.server.healthcheck;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.util.List;
+
+public class DefaultMemoryHealthChecker implements HealthChecker {
+
+    private final List<MemoryPoolMXBean> memoryPoolMXBeans;
+    private final double targetMemoryHeapUsage;
+    private final double targetMemoryNonHeapUsage;
+    private final double targetMemoryTotalUsage;
+
+    public DefaultMemoryHealthChecker(final double targetMemoryHeapUsage, final double targetMemoryNonHeapUsage, final double targetMemoryTotalUsage) {
+        this.targetMemoryHeapUsage = targetMemoryHeapUsage;
+        this.targetMemoryNonHeapUsage = targetMemoryNonHeapUsage;
+        this.targetMemoryTotalUsage = targetMemoryTotalUsage;
+        this.memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans();
+    }
+
+    @Override
+    public boolean isHealthy() {
+        final double heapMemoryUsage = getHeapMemoryUsage();
+        final double nonHeapMemoryUsage = getNonHeapMemoryUsage();
+        final double totalMemoryUsage = getTotalMemoryUsage();
+        return heapMemoryUsage <= targetMemoryHeapUsage &&
+               nonHeapMemoryUsage <= targetMemoryNonHeapUsage &&
+               totalMemoryUsage <= targetMemoryTotalUsage;
+    }
+
+    private double getHeapMemoryUsage() {
+        return this.memoryPoolMXBeans.stream().filter(e -> MemoryType.HEAP.equals(e.getType())).map(e -> e.getUsage().getUsed()).mapToDouble(e -> e).sum();
```

Comment:
According to its Javadoc, `ManagementFactory.getMemoryPoolMXBeans()` can return a different set of beans on each call. Can we call `MAnagementFactory.getMemoryPoolMXBeans()` here instead?

Reply:
1a7bfbbe6a554e4dead2462415c7a3ee431c807c I changed it so that it calls ManagementFactory.getMemoryPoolMXBeans() every time.